### PR TITLE
feat: search all demos

### PR DESCRIPTION
**Pull request by s-pace**
This PR helps to outbound project by showing them how the discovery experience will look like for their website.

One can choose to use either the v2 or v3 version. Adding a `GET` parameter `?v3=true` will enforce the rendering with the last version of the UI.

One can also set up the index to use by providing as `GET` parameters the following information:
`?indexName=<yourIndexName>&apiKey=<yourAPIKey>`

User can also retrieve an existing demo by using the dedicated element which will search on our list of available demos.

### CODE REVIEW

**Comment by s-pace** [comment]
I want to attract people with this search when we contact ppl by support

File: src/pages/demo.js [hunk 1]
```diff
@@ -180,7 +240,42 @@ function Demo() {
...
+              id="searchButton"
+              primary
+            >
+              Open modal to search on {project.indexName}
+            </Button>
+            {isOpen &&
+              createPortal(
+                <DocSearchModal
+                  {...project}
+                  navigator={{
```
Comment (francoischalifour) [inline src/pages/demo.js]: For links to effectively work, we need to remove `navigator`, `transformItems` and `hitComponent`.

Comment (s-pace) [reply src/pages/demo.js]: That was my next question. Should be done

**Review by Shipow — APPROVED** [review]

**Comment by s-pace** [comment]
TODO:

- [ ] Filter on latest version for each demo

**Pull request closed without merging**